server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/SnapshotShardsStats.java

```diff
@@ -128,4 +141,87 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
+        XContentParser.Token token;
+        int initializingShards = 0;
+        int startedShards = 0;
+        int finalizingShards = 0;
+        int doneShards = 0;
+        int failedShards = 0;
+        int totalShards = 0;
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                String currentName = parser.currentName();
+                if (currentName.equals(Fields.INITIALIZING)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    initializingShards = parser.intValue();
+                } else if (currentName.equals(Fields.STARTED)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    startedShards = parser.intValue();
+                } else if (currentName.equals(Fields.FINALIZING)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    finalizingShards = parser.intValue();
+                } else if (currentName.equals(Fields.DONE)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    doneShards = parser.intValue();
+                } else if (currentName.equals(Fields.FAILED)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    failedShards = parser.intValue();
+                } else if (currentName.equals(Fields.TOTAL)) {
+                    if (parser.nextToken() != XContentParser.Token.VALUE_NUMBER) {
+                        throw new ElasticsearchParseException("failed to parse snapshot shards stats, expected number for field [{}]",
+                            currentName);
+                    }
+                    totalShards = parser.intValue();
```

Comment:
I wonder if it's necessary to throw specific errors for each case here. It is nice to have good error messages, on the other hand these would be thrown anyways (I think) by the parser, maybe a bit more cryptic. But most importantly, is it worth it given that we are parsing our own output. Certainly we are not going to change the type of these fields one way?